modules/markup/html.go

```diff
@@ -657,6 +661,12 @@ func sha1CurrentPatternProcessor(ctx *postProcessCtx, node *html.Node) {
 	// but that is not always the case.
 	// Although unlikely, deadbeef and 1234567 are valid short forms of SHA1 hash
 	// as used by git and github for linking and thus we have to do similar.
+	// Because of this, we check to make sure that a matched hash is actually
+	// a commit in the repository before making it a link.
+	if _, err := git.NewCommand("rev-parse", "--verify", hash).RunInDirBytes(ctx.metas["repoPath"]); err != nil {
+		return
```

Comment:
```suggestion
		 log.Error("sha1CurrentPatternProcessor git rev-parse: %v", err)
		return
```
Should log this error I think

Reply:
Done. Thanks.

Reply:
Actually, on second thought, maybe this should not be logged. The error will occur anytime someone loads a page that is rendered using this function that contains something that looks like a hash but isn't. That could make far too much noise in the log.

Reply:
Hmm fair enough, though we should be able to log an error that isn't just a failed match.

It looks like other parts of the code do something like this to avoid expected errors but log unexpected ones when using git.NewCommand:

```
if err != nil && !strings.HasPrefix(err.Error(), "exit status 128 - fatal: Needed a single revision") {
log.Error ....
```
Can you test something like that?

Reply:
I think it is ok not to log this as this will not result in user error in any way, worst case is that no link will be rendered

Reply:
Mostly for debugging, as this is a big change and if somebody writes in and says "none of these turn into links anymore" we will have nothing to suggest to them

Reply:
log.Debug would be ok but log.Error imho is way to big level for this

Reply:
Okay, `log.Debug` is used to log errors that were not the expected `fatal: Needed a single revision`.

Reply:
I guess the logic is that actual errors that are logged with the code above should be rare and are probably an indication of a larger problem (if git.NewCommand gives an odd error there is probably something wrong with the repo, git, permissions, or overall logic. It wouldn't be an error related to hash links or anything like that). Other uses of git.NewCommand seem to use log.error in places.

Anyway I don't have a strong opinion other than we should log it because when anything doesn't work the first thing we tell users is to look in gitea.log and it's always hard when nothing is there. Otherwise lgtm